core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -855,6 +901,8 @@ def hyperparameter_tune(self, scheduler_options, time_limit=None, **kwargs):
         if 'time_out' not in scheduler_options[1]:
             scheduler_options[1]['time_out'] = time_limit
         kwargs = self.initialize(time_limit=scheduler_options[1]['time_out'], **kwargs)
+        self._register_fit_metadata(**kwargs)
+        self._validate_fit_memory_usage(**kwargs)
```

Comment:
This code is not being added in `get_features` (Notice there is a code block separation, this is in `hyperparameter_tune`. We call them in `hyperparameter_tune` so we can raise NotEnoughMemoryError before starting HPO rather than raising on each trial.

Reply:
My bad.